python/hsfs/storage_connector.py

```diff
@@ -1089,6 +1089,10 @@ def read(
         # Returns
             `Dataframe`: A Spark dataframe.
         """
+        if not engine.get_instance().is_connector_type_supported(StorageConnector.GCS):
```

Comment:
Maybe this is cleaner
```suggestion
        if not engine.get_instance().is_connector_type_supported(self.type):
```